scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -2087,42 +2087,55 @@ class ScalametaParser(input: Input, dialect: Dialect) { parser =>
       atPos(t, auto)(Term.Match(t, inBracesOrNil(caseClauses())))
     }
   }
+
+  def ifClause(isInline: Boolean = false) = {
+    accept[KwIf]
+    val (cond, thenp) = if (token.isNot[LeftParen] && dialect.allowSignificantIndentation) {
+      val cond = expr()
+      acceptOpt[LF]
+      accept[KwThen]
+      (cond, exprMaybeIndented())
+    } else {
+      val forked = in.fork
+      var cond = condExpr()
+      if ((token.is[Ident] && isLeadingInfixOperator(token)) || token.is[Dot]) {
+        in = forked
+        cond = expr()
+      }
+      newLinesOpt()
+      acceptOpt[KwThen]
+      (cond, exprMaybeIndented())
+    }
+
+    val ifExpr = if (tryAcceptWithOptLF[KwElse]) {
+      Term.If(cond, thenp, exprMaybeIndented())
+    } else if (token.is[Semicolon] && ahead { token.is[KwElse] }) {
+      next(); next(); Term.If(cond, thenp, expr())
+    } else {
+      Term.If(cond, thenp, atPos(in.tokenPos, in.prevTokenPos)(Lit.Unit()))
+    }
+
+    if (isInline) {
+      ifExpr.setMods(List(Mod.Inline()))
+    }
+    ifExpr
+  }
+
   // FIXME: when parsing `(2 + 3)`, do we want the ApplyInfix's position to include parentheses?
   // if yes, then nothing has to change here
   // if no, we need eschew autoPos here, because it forces those parentheses on the result of calling prefixExpr
   // see https://github.com/scalameta/scalameta/issues/1083 and https://github.com/scalameta/scalameta/issues/1223
   def expr(location: Location, allowRepeated: Boolean): Term =
     autoPos(token match {
+      case SkInline() if ahead(token.is[KwIf]) =>
```

Comment:
Turns out there was one more issue with ifs, so a much safer place to check it is here.

The problem was with
`def a = inline if cond then a`